### ISSUE TITLE
Skip and proceed in the case of CorruptRecordException

### ISSUE THREAD

**Issue by vineetgoel** [aio-libs/aiokafka#321]
I get the following error while consuming and then consumption stops

```
    records = res_or_error.getall(max_records)
  File "/home/robinhood/env/lib/python3.6/site-packages/aiokafka/consumer/fetcher.py", line 100, in getall
    for msg in chain([first_msg], next_batch_iter):
  File "/home/robinhood/env/lib/python3.6/site-packages/aiokafka/consumer/fetcher.py", line 850, in _unpack_records
    raise Errors.CorruptRecordException("Invalid CRC")
kafka.errors.InvalidMessageError: [Error 2] InvalidMessageError: Invalid CRC
```

I wonder if there should be a flag to inform the consumer to keep going forward in the case of a corrupted message instead of getting stuck.

**Comment by tvoinarovskyi** [comment]
Kind of curious how Java client reacts here...

**Comment by tvoinarovskyi** [comment]
Seems like Java client also just raises an error to the user. https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java#L1117
How did you even get that case? I mean the broker checks CRC32 sums on produce...

**Comment by tvoinarovskyi** [comment]
It is a pain thou, as you don't know what failed a batch or a message and how much offsets to skip. All you can do is seek() 1 position forward on the error really...

**Comment by tvoinarovskyi** [comment]
There is an option to ignore CRC check completely thou.

**Comment by vineetgoel** [comment]
What happens if I ignore the CRC check? Will things just crash or will it move on until valid messages are found?

Interestingly, we are producing using aiokafka as well as consuming using it. Is there some broker/consumer/producer configuration I should be aware of?

**Comment by tvoinarovskyi** [comment]
In case you disable the CRC check it may not crash, as long as no broken data is present. The broker validates records and does iteration over them, so the record structure itself should be readable. Probably this is a result of some bug, but I have no idea where and how this happened.
For what I can add:
* Broker will validate the CRC on produce
* Broker will not validate on consume, so the data got corrupted on disk somehow (?).

What Broker version are you using? And what message format?

**Comment by vineetgoel** [comment]
I am using 0.10.2.1 and how can I check my message format? I don't change any of the producer/consumer defaults. The messages are all in json.

Coincidentally, looks like I am seeing this error only on topics that were produced into by the aiokafka producer. Any ideas on how I can go about troubleshooting this?

**Comment by bt-wil** [comment]
What is the option to ignore the CRC check?  I haven't ran into this issue, but I want the consumer to resilient in that it doesn't crash and stops.  Currently, I would just drop the record if there is any CRC errors.

I'm using the `getmany()` method to retrieve records.  If there is a CRC error, would that throw an exception?  Would calling `getmany()` again retains the same offsets and basically get 'stuck'?  Can I get around this?

**Comment by vineetgoel** [comment]
@tvoinarovskyi is there a way to change the format of message the producer is sending. We have a hypothesis that the LegacyRecord format might not be playing too well.

**Comment by tvoinarovskyi** [comment]
Not with the public API. What exactly is wrong with the message?

**Comment by tvoinarovskyi** [comment]
Are the errors posten on `kafka-python` from those topics?

**Comment by vineetgoel** [comment]
Yeah! Not sure if anything is wrong with the message. Essentially, ever since we upgraded aiokafka to 0.4.0 we are seeing a lot more of these errors. Not sure if they are related though. There is probably something weird happening with the broker. But out of curiosity was there any major changes in aiokafka 0.4.0 with regards to message production rate/message format?

**Comment by tvoinarovskyi** [comment]
Yes, it's rewritten fully.
> Rewrote Records parsing protocol to allow implementation of newer protocol versions later

**Comment by vineetgoel** [comment]
Can you think of anything in the message format that might result in message corruption on disk?

**Comment by tvoinarovskyi** [comment]
Relates to https://github.com/dpkp/kafka-python/issues/1409

**Comment by tvoinarovskyi** [comment]
@vineetgoel For information:
 * you use the C extension, or Python implementation of the parser? If `AIOKAFKA_NO_EXTENSIONS` environment is not provided, C extension is used.
 * Are you using compression on the topic?
 * Is the topic compacted?

**Comment by tvoinarovskyi** [comment]
@vineetgoel Just to confirm, your not producing enormous messages, do you? Like a few GB =)

**Comment by tvoinarovskyi** [comment]
Just as a wild guess you can try to disable the C extension by setting `AIOKAFKA_NO_EXTENSIONS=1` environment variable. Maybe some strange bug in C code...

**Comment by tvoinarovskyi** [comment]
Just as a wild guess you can try to disable the C extension by setting `AIOKAFKA_NO_EXTENSIONS=1` environment variable. Maybe some strange bug in C code...

**Comment by vineetgoel** [comment]
@tvoinarovskyi :
- Probably using the C extension as I haven't added any environment variables or explicitly set any such settings
- Not using compression on the topics
- Yes, the topic is compacted. Could this affect anything?
No, the messages should be fairly small. Nothing out of the ordinary there.

**Comment by vineetgoel** [comment]
Is the C extension a new thing?

**Comment by vineetgoel** [comment]
I see! Cool, thanks :)

**Comment by tvoinarovskyi** [comment]
@vineetgoel Did you manage to get any new info on the matter. I could not reproduce this in any of my setups, with neither compacted nor not compacted topics. The broker will neither accept nor corrupt existing data on compaction.
Tried different length ingesting, padding bytes, but it still works as expected. From what I could find the only real source of those errors is an actual disk failure. I don't know why would upgrading to aiokafka 0.4 enforce more errors, maybe it has something to do with speed enhancements.
